internal/services/compute/client/client.go

```diff
@@ -251,8 +251,8 @@ func NewClient(o *common.ClientOptions) (*Client, error) {
 	}, nil
 }
 
-func (c *Client) CancelRollingUpgradesBeforeDeletion(ctx context.Context, id parse.VirtualMachineScaleSetId) error {
-	resp, err := c.VMScaleSetRollingUpgradesClient.GetLatest(ctx, id.ResourceGroup, id.Name)
+func (c *Client) CancelRollingUpgradesBeforeDeletion(ctx context.Context, id virtualmachinescalesets.VirtualMachineScaleSetId) error {
```

Comment:
Whilst this works, since VMSS (and VM) IDs are so widely used, we should add the Resource IDs to [`commonids` within `hashicorp/go-azure-helpers`](https://github.com/hashicorp/go-azure-helpers/tree/main/resourcemanager/commonids) and then (once that's merged/vendored into `hashicorp/go-azure-sdk` and the provider) [add the new Common IDs to Pandora so that we can use those](https://github.com/hashicorp/pandora/tree/main/tools/importer-rest-api-specs/components/parser/resourceids) instead of outputting these to each package.

As such, would you mind sending those two PR's first to set this up as a Common ID? Otherwise we'll need to update large chunks of the codebase each time we bump the Compute API (which happens fairly often) and will cause merge conflicts

Reply:
Added a guide too fwiw: https://github.com/hashicorp/pandora/pull/3024